File: Data Structures & Algorithms/jump-game/submission-0.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        p1 = len(nums) - 1
        p2 = p1 - 1

        i = 1
        while p1 >= 0 and p2 >= 0:
            print(p2, p1)
            
            if nums[p2] >= i:
                p1 = p2
                p2 = p1 - 1
                i = 1
                continue
            
            p2 -= 1
            i += 1
        print(p2, p1)
        return p1 == 0
        

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    productExceptSelf(nums) {
        let output = [];
        for(let i = 0; i < nums.length; i++)
        {
            let left = 1;
            let right = 1;

            for(let j = 0; j < nums.length; j++)
            {
                if(i == j) continue;
                if(j < i) {
                    left *= nums[j];
                }
                else {
                    right *= nums[j];
                }
            }

            output.push(left * right);
        }
        return output;
    }
}
